source/gitidentifier.go

```diff
@@ -2,29 +2,32 @@ package source
 
 import (
 	"net/url"
+	"regexp"
 	"strings"
 
 	"github.com/pkg/errors"
 )
 
+// sshGitRegexp is used to detect if the git repo uses ssh
+// e.g. git@... or otheruser@nonstandardgithost.com:my/really/strange/repo.git
+var sshGitRegexp, _ = regexp.Compile("[a-z0-9_]+@[^/]+:.+")
```

Comment:
Not sure if these changes are related to the rest of the PR. The identifier logic is copied from Docker description https://github.com/moby/moby/blob/master/pkg/urlutil/urlutil.go#L21 and need to be aware of the ambiguity concerns (eg. `@` could be valid for HTTP as well) and maybe change docs. I'd rather handle that separately. @thaJeztah

Reply:
That's correct that it's not related to mounting the ssh socket, i'll back this change out in this PR, then open a new one to tackle the issue of using a non-git username for ssh cloning.